providers/infrastructure-vsphere/v1.5.1/cconly/base.yaml

```diff
@@ -2100,3 +2103,56 @@ spec:
             {{- range .additionalFQDN }}
             - {{ . }}
             {{- end }}
+  - name: systemWideProxy
+    enabledIf: '{{ .proxy.systemWide }}'
+    definitions:
+    - selector:
+        apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+        kind: KubeadmControlPlaneTemplate
+        matchResources:
+          controlPlane: true
+      jsonPatches:
+      - op: add
+        path: /spec/template/spec/kubeadmConfigSpec/preKubeadmCommands/-
+        value: echo "export HTTP_PROXY={{- .proxy.httpProxy }}" >> /etc/profile
```

Comment:
Since the command contains template, so can't directly add by value, it should be 
```
valueFrom:
          template: |
            echo "export HTTPS_PROXY= {{- .proxy.httpsProxy }}" >> /etc/profile
```